components/telemetry-operator/controllers/logpipeline_controller.go

```diff
@@ -52,13 +52,14 @@ type LogPipelineReconciler struct {
 }
 
 // NewLogPipelineReconciler returns a new LogPipelineReconciler using the given FluentBit config arguments
-func NewLogPipelineReconciler(client client.Client, scheme *runtime.Scheme, daemonSetConfig sync.FluentBitDaemonSetConfig, emitterConfig fluentbit.EmitterConfig) *LogPipelineReconciler {
+func NewLogPipelineReconciler(client client.Client, scheme *runtime.Scheme, daemonSetConfig sync.FluentBitDaemonSetConfig, emitterConfig fluentbit.EmitterConfig, fluentBitMaxFSBufferSize string) *LogPipelineReconciler {
```

Comment:
Maybe let's make the EmitterConfig a more general "PipelineConfig" and have the max. filesystem buffer size in there. Then we would not grow this argument list to infinity ;-) Maybe you can find a better name than "PipelineConfig".